src/System.Private.ServiceModel/tools/IISHostedWcfService/App_code/VerifyWebSockets.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System;
+using System.ServiceModel;
+
+namespace WcfService
+{
+    [ServiceBehavior(InstanceContextMode = InstanceContextMode.PerSession, AddressFilterMode = AddressFilterMode.Any)]
+    public class VerifyWebSockets : IVerifyWebSockets
+    {
+        private String successMessage = "";
+
+        // This operation cannot return anything so in order to validate that it was called set the InstanceContextMode
+        //  to PerSession and set a variable that we can get with a call to the ValidateWebSocketsUsed operation.
+        public void ForceWebSocketsUse()
+        {
+            successMessage = "WebSocketsWasUsed";
```

Comment:
Nit: not blocking, but I generally like to decouple the service from the test as much as possible, including magic strings the server side returns.  I'd rather do something like pass a String into ForceWebSocketsUse, store it as you do here, and return it in ValidateWebSocketsUsed.  That way the magic string "WebSocketsWasUses" will exist only in the test.